crates/torii/server/src/proxy.rs

```diff
@@ -103,15 +105,19 @@ impl Proxy {
                         .collect::<Vec<HeaderName>>(),
                 );
 
-            let cors = match allowed_origins.as_slice() {
-                [origin] if origin == "*" => cors.allow_origin(AllowOrigin::mirror_request()),
-                origins => cors.allow_origin(
-                    origins.iter().map(|o| o.parse().expect("valid origin")).collect::<Vec<_>>(),
-                ),
-            };
+            let cors =
+                allowed_origins.clone().map(|allowed_origins| match allowed_origins.as_slice() {
+                    [origin] if origin == "*" => cors.allow_origin(AllowOrigin::mirror_request()),
+                    origins => cors.allow_origin(
+                        origins
+                            .iter()
+                            .map(|o| o.parse().expect("valid origin"))
+                            .collect::<Vec<_>>(),
+                    ),
+                });
 
             let graphql_addr_clone = graphql_addr.clone();
-            let service = ServiceBuilder::new().layer(cors).service_fn(move |req| {
+            let service = ServiceBuilder::new().option_layer(cors).service_fn(move |req| {
```

Comment:
`cors` is never `None`. We may want to add a `if` to ensure `None` is passed to this function if `allowed_origins` is None.

Reply:
wdym? the second definition of `let cors` maps the allowed_origins which is an Option<Vec<String>> so it should be None if there are no provided origins

Reply:
Oh, my bad by default it's `*`. Then how can I pass `None`?
I guess by default it's better to have `None` now that we can disable it?